.github/workflows/release.yml

```diff
@@ -41,6 +41,16 @@ jobs:
         os: ['macos-latest', 'windows-latest', 'ubuntu-18.04']
         py-version: ['3.6', '3.7', '3.8', '3.9']
         tf-version: ['2.4.2', '2.6.0']
+        cpu: ['x86']
+        include:
+          - os: 'macos-11'
+            cpu: 'arm64'
+            tf-version: '2.5.0'
+            py-version: '3.8'
+          - os: 'macos-11'
+            cpu: 'arm64'
+            tf-version: '2.5.0'
+            py-version: '3.9'
```

Comment:
[`tensorflow-macos`](https://pypi.org/project/tensorflow-macos/#files) currently only supports TF 2.5.0 and Python 3.8 and 3.9.

Reply:
And tensorflow-metal?

Reply:
As far as I can tell `tensorflow-metal` is an optional dependency to enable metal support via the new pluggable device mechanism so it is not really necessary here since as far as I know TFA doesn't directly link against the metal plugin.

Reply:
Yes I meant if It Is constrained to 2.5.0 now I suppose that all the OSX stuffs are aligned on 2.5.0 including GPU acell

Reply:
Yes, `tensorflow-metal` is also only built against 2.5.0

Reply:
I am not sure how the `tensorflow-macos` wheels are generated, so there is not really a way to see whether 2.6 support is right around the corner or not.

Reply:
What I see is that `tensorflow-macos` [is archived](https://github.com/apple/tensorflow_macos).
`tensorflow-metal` will work with official Apple conda channel [tensorflow-deps](https://anaconda.org/apple/tensorflow-deps/files) release but what it is strange is that deps has already `2.6.0`.

In the official documentation `tensorflow-deps` it is an official step also for M1:

https://developer.apple.com/metal/tensorflow-plugin/

Reply:
> What I see is that tensorflow-macos is [archived](https://github.com/apple/tensorflow_macos).

That is correct. As far as I can tell, this repo was used to [build wheels based on TF 2.4.rc0](https://github.com/apple/tensorflow_macos#supported-versions).

Step two in https://developer.apple.com/metal/tensorflow-plugin/ says to install `tensorflow-macos` which seems like it is built directly from the main https://github.com/tensorflow/tensorflow repo. Unfortunately there is not really a way to verify where the wheels are coming from.

> tensorflow-metal will work with the official Apple conda channel tensorflow-deps release but it is strange that tensorflow-deps has already 2.6.0

It looks like `tensorflow-deps` has already 2.6.0 was only released yesterday, so maybe they will update `tensorflow-macos` soon as well. As far as I know the `tensorflow-deps` package only includes the pip dependencies from TensorFlow built for ARM. Since we only need to have the prebuilt `*.so` file from TensorFlow available to link against we don't need to install `tensorflow-deps`. We cannot run it on the x86 GitHub actions VM anyway. Or am I missing something?

Reply:
> Unfortunately there is not really a way to verify where the wheels are coming from.
Yes as  I think you already know TF CI is not as easy like to collaborate on Github Actions on the repo.

`tensorflow-macos` is still owned by Google and are the macos  ARM64 and X86_64 builds maintained by @kulinseth

Instead in the standard `tensorflow` wheel releases have ony macos X86_64 and no ARM64 coverage but `2.6.0` is available as on other platforms

> tensorflow-deps has already 2.6.0 was only released yesterday,

Yes this was just the point for guessing if `tensorflow-macos`  was imminent or not before we merge this PR but I think that only @kulinseth could help on this as he is the maintainer of both `tensorflow-macos` and `tensorflow-metal` on pypi.

If the release is imminent we could just wait and edit this PR, if not I think that we could merge and open a new one later.

Reply:
> `tensorflow-macos` is still owned by Google and are the macos ARM64 and X86_64 builds maintained by @kulinseth

Oh this is interesting, I didn't know that. But then, what is the reason why the Python 3.8 and 3.9 ARM wheels are not published under the `tensorflow` in the first place? From the outside it looks like the `*macosx_11_0_arm64.whl` wouldn't conflict with any of the existing wheels.

Reply:
Cause I think that `tensorflow` pypi macos releases are produced on the Google build infra  and SIG build ownership with:

https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/ci_build/rel/macos

Instead  `tensorflow-macos` builds that include ARM64 I think are managed by @kulinseth on external CI/Build resources

Reply:
> Instead tensorflow-macos builds that include ARM64 I think are managed by @kulinseth on external CI/Build resources

I see, although I don't think PyPI requires all wheels to be uploaded at the same time, so if the external CI would have permissions to write to `tensorflow` directly this shouldn't be a problem since the ARM wheels won't conflict with any of the existing wheels already uploaded by the SIG build infra.

Reply:
Yes, as I told you it is not about conflicts, It is about that if `tensorflow-macos` is imminent to land on 2.6.0 as `tensorflow-deps` is ready for TF 2.6.0 I will just wait to edit this PR directly to have a 2.6.0 wheel published for TF addons.
If not probably we will merge as is.

Reply:
> Yes, as I told you it is not about conflicts, It is about that if `tensorflow-macos` is imminent to land on 2.6.0 as `tensorflow-deps` is ready for TF 2.6.0 I will just wait to edit this PR directly to have a 2.6.0 wheel published for TF addons.
> If not probably we will merge as is.

@mihaimaruseac and @penpornk
 
Thanks @bhack for the heads-up. Is this PR for enabling ARM64 builds for TF through official build channels which will be hosted on pypi ?

Reply:
>  Is this PR for enabling ARM64 builds for TF through official build channels which will be hosted on pypi ?

No this is limited for Tensorflow Addons ARM64 builds in the official Tesnroflow Addons pypi so we need to know if we could enable 2.6.0 for `tensorflow-macos` 2.6.0.

> ARM64 builds for TF through official build channels which will be hosted on pypi 

For this you really need a brand new PR in the TF own repository over:
(Stable) 
https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/ci_build/rel/macos
(Eventually nightly) https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/ci_build/nightly_release/macos


But it is another topic.

Reply:
@bhack , thanks for the update. We are working on the v2.6 release, I will update here regarding that. I see this PR has merged we can adopt to `tensorflow-macos` after we make the release in a separate PR.

Reply:
@kulinseth Are you interested to manage the TF addons wheels release on your pypi on a new `tensorflow-macos/metal` release?